packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -736,6 +736,10 @@ class BaseSynchronizedData:
     This is the relevant data constructed and replicated by the agents.
     """
 
+    # Keys always set by default
+    # TODO: round_count and period_count need to be guaranteed to be synchronized too.
+    default_db_keys: List[str] = ["round_count", "period_count", "nb_participants"]
```

Comment:
`nb_participants` does not exist in the db. Instead, it is inferred from the `participants`. I think that `round_count`, `period_count`, `participants`, and `all_participants` should be `default_db_keys` and also they should always be pre and post-conditions. Therefore, we might need to simplify the code and make it optional for the FSM to specify these as pre and post-conditions and automatically assume it and fail if that is not the case.

Reply:
I see you already suggested what I pointed out in https://github.com/valory-xyz/open-autonomy/pull/1659#discussion_r1047713895.

Reply:
`nb_participants` is a weird case. I suggest we keep it as a default pre condition as it defaults to 0 when no `participants` are present.

Reply:
Agreed! Let's keep it in mind for `all_participants` to add them as a default key in the future.